datajunction-clients/python/datajunction/admin.py

```diff
@@ -21,19 +21,24 @@ def get_catalog(self, name: str) -> dict:
         response = self._session.get(f"/catalogs/{name}/", timeout=self._timeout)
         return response.json()
 
-    def add_catalog(self, name: str) -> None:
+    def add_catalog(self, name: str, skip_if_exists: bool = False) -> None:
         """
         Add a catalog.
         """
-        response = self._session.post(
-            "/catalogs/",
-            json={"name": f"{name}"},
-            timeout=self._timeout,
-        )
-        json_response = response.json()
+        try:
+            response = self._session.post(
+                "/catalogs/",
+                json={"name": f"{name}"},
+                timeout=self._timeout,
+            )
+        except DJClientException as exc:  # pragma: no cover
+            if skip_if_exists and "already exists" in str(exc):
```

Comment:
Should this have some message saying that it already exists so that users are informed?